### DISCUSSION TITLE
Circular images as markers on the map

### DISCUSSION THREAD

**Discussion by am0ka** [rnmapbox/maps#3104] [Q&A]
Hello community, I am here to ask for help. I am trying to implement circular image marker with colored border. Up to now I have 2 options, and both of them has their flaws, that I can't solve myself:

1. Use of ShapeSource, CircleLayer and SymbolLayer, to draw multiple markers.
2. Use of MarkerView to make the same thing.

Problem with first option is that image is always rectangular and I can not fit it inside circle, I am not so good with MapBox scripting
Problem with second option, is that MarkerView is not clickable, I am putting Pressable/TouchableOpacity as first direct child to MarkerView but nothing is working.

I need help whether with MarkerView to make it clickable, or to make picture in SymbolLayer apply rouded form

Here is how it looks like, top is option 1, and bottom is option 2:
<img src="https://i.ibb.co/qYSzXdH/Screenshot-2023-10-13-at-17-12-00.png" alt="Screenshot-2023-10-13-at-17-12-00" border="0"><img src="https://i.ibb.co/t8Lk6TR/Screenshot-2023-10-13-at-17-14-22.png" alt="Screenshot-2023-10-13-at-17-14-22" border="0">

Here is code implementation of both variants:
1. 
```
<Mapbox.ShapeSource
  id="markers"
  onPress={async (e) => {
    // if (!shapeSourceRef.current) return;
    const [feature] = e.features;
    // handleMarkerPress(feature as MarkerData);
    console.log({
      feature,
      coordinates: e.coordinates,
    });
  }}
  hitbox={{ width: 5, height: 5 }}
  shape={{
    type: "FeatureCollection",
    features: markers.map((m) => ({
      type: "Feature",
      id: `${m?.properties.id}_${
        m.properties.is_thread ? "thread" : "marker"
      }`,
      properties: { ...m?.properties },
      geometry: {
        type: "Point",
        coordinates: m?.geometry.coordinates,
      },
    })),
  }}
>
  <Mapbox.CircleLayer
    id="marker"
    style={{
      circleColor: "transparent",
      circleOpacity: 1,
      circleStrokeWidth: 4,
      circleStrokeColor: [
        "case",
        ["boolean", ["get", "is_thread"], false],
        tw.color("cyan-300"),
        tw.color("emerald-300"),
      ],
      circleSortKey: ["get", "is_thread"],
      circleRadius: 30,
      circlePitchAlignment: "map",
    }}
  />
  <Mapbox.SymbolLayer
    belowLayerID="marker"
    id="title"
    style={{
      iconImage: "{id}",
      iconAllowOverlap: true,
      iconSize: 0.95,
    }}
  />
</Mapbox.ShapeSource>
```
2.
```
{
markers.map(({{ coordinate, id, imageUrl, isThread, onPress }}) => (
<Mapbox.MarkerView coordinate={coordinate} allowOverlap>
  <TouchableOpacity
    style={tw`flex items-center justify-center`}
    onPress={() => onPress()}
  >
    <View
      style={[
        tw`w-15 h-15 bg-white border-4 `,
        isThread
          ? tw`rounded-2xl border-emerald-300`
          : tw`rounded-full border-cyan-300`,
      ]}
    >
      <FastImage
        style={[
          tw`w-full h-full`,
          isThread ? tw`rounded-xl` : tw`rounded-full`,
        ]}
        source={{ uri: imageUrl }}
        resizeMode="cover"
      />
    </View>
    <View
      style={[
        tw`h-2.5 w-2.5 mb-18 mt-1.5 rounded-full`,
        isThread ? tw`bg-emerald-300` : tw`bg-cyan-300`,
      ]}
    />
  </TouchableOpacity>
</Mapbox.MarkerView>))
}
```

**Comment by joostwmd** [comment]
Hey, 

i solved this one: you can crop the image to a circle. I used sharpJS for this. Then you can get the base64 string from the cropped image. And that string you can use to display your custom icon (the Mapbox.Images accepts a base64 string).

Hope this helps, if you need more detailed info, please let me know
